lib/stages/visTranspose.cpp

```diff
@@ -227,22 +233,32 @@ void visTranspose::main_thread() {
     uint64_t frame_size = 0;
 
     // wait for a non-empty frame to get dataset ID from
-    uint32_t first_ind = 0;
+    uint32_t num_empty_skip = 0;
     while (true) {
         // Wait for a frame in the input buffer in order to get the dataset ID
-        if ((wait_for_full_frame(in_buf, unique_name.c_str(), first_ind)) == nullptr) {
+        if ((wait_for_full_frame(in_buf, unique_name.c_str(), frame_id)) == nullptr) {
             return;
         }
-        auto frame = VisFrameView(in_buf, first_ind);
+        auto frame = VisFrameView(in_buf, frame_id);
+
+        // Get the frame size to publish metrics later
+        if (frame_size == 0)
+            frame_size = frame.calculate_buffer_layout(num_input, num_prod, num_ev).first;
+
+        // If the frame is empty, release the buffer and continue
         if (frame.fpga_seq_total == 0 && frame.dataset_id == dset_id_t::null) {
-            INFO("Got empty frame ({:d}).", first_ind);
-            first_ind++;
+            mark_frame_empty(in_buf, unique_name.c_str(), frame_id++);
+            num_empty_skip++;
         } else {
             ds_id = frame.dataset_id;
             break;
         }
     }
 
+    if (num_empty_skip > 0) {
+        INFO("Found {:d} empty frames at the start of the file.", num_empty_skip);
```

Comment:
I think this will be useful to have present in the gossec logs. Logging every empty frame completely overwhelms the logs, so this is meant as a summary.

Reply:
Wouldn't you want the summary also if the empty frames are not at the beginning of the file, though?

Reply:
the empty frames at the beginning of the file are the special case that was tripping gossec up. So I think these are separate cases, and I'm much more interested in knowing about the frames at the start. I guess this is not a crucial message, so I can remove it. But it can only be printed once per file.

Reply:
Hm, guess I don't understand how after fixing this bug an empty frame at the beginning of the file is more interesting than in another place.

Reply:
yeah I guess it's not super relevant, it just confirms that gossec successfully dealt with this file. In my mind it's performing an 'exceptional' action by dropping all these frames at the start so some sort of record is useful